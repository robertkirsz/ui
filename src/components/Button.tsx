import styled, { css } from 'styled-components'

export type Props = {
  primary?: boolean
  size?: 'small' | 'medium' | 'large'
}

export default styled.button<Props>`
  border: none;
  cursor: pointer;

  ${({ primary = false, size = 'medium', theme }) => css`
    color: ${primary ? 'white' : theme.colors.text};
    border-radius: ${theme.borderRadius};
    background: ${primary ? theme.colors.primary : theme.colors.secondary};

    ${!primary && 'box-shadow: rgba(0, 0, 0, 0.15) 0 0 0 1px inset;'}

    ${size &&
    {
      small: 'font-size: 12px; padding: 10px 16px;',
      medium: 'font-size: 14px; padding: 11px 20px;',
      large: 'font-size: 16px; padding: 12px 24px;'
    }[size]}
  `}
`
