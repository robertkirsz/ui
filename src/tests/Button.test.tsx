import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Button from '../components/Button'

describe('Button', () => {
  it('Is a button', () => {
    const { getByText } = render(<Button>Button</Button>)
    const element = getByText('Button')

    expect(element).toBeInTheDocument()
    expect(element.tagName.toLowerCase()).toBe('button')
  })

  it('Is clickable', () => {
    const clickCallback = jest.fn()
    const { getByText } = render(<Button onClick={clickCallback}>Button</Button>)

    fireEvent.click(getByText('Button'))
    expect(clickCallback).toBeCalled()
  })

  it('Matches snapshots', () => {
    const { getByTestId } = render(
      <div data-testid="Container">
        <Button primary>Button</Button>
        <Button>Button</Button>
        <Button size="small">Button</Button>
        <Button size="medium">Button</Button>
        <Button size="large">Button</Button>
      </div>
    )

    expect(getByTestId('Container')).toMatchSnapshot()
  })
})
