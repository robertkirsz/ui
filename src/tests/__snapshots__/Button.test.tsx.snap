// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Matches snapshots 1`] = `
.c0 {
  border: none;
  outline: none;
  font: inherit;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: white;
  border-radius: 4px;
  background: #1ea7fd;
  font-size: 14px;
  padding: 11px 20px;
}

.c1 {
  border: none;
  outline: none;
  font: inherit;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #333;
  border-radius: 4px;
  background: none;
  font-size: 14px;
  padding: 11px 20px;
  box-shadow: rgba(0,0,0,0.15) 0 0 0 1px inset;
}

.c2 {
  border: none;
  outline: none;
  font: inherit;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #333;
  border-radius: 4px;
  background: none;
  font-size: 12px;
  padding: 10px 16px;
  box-shadow: rgba(0,0,0,0.15) 0 0 0 1px inset;
}

.c3 {
  border: none;
  outline: none;
  font: inherit;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #333;
  border-radius: 4px;
  background: none;
  font-size: 16px;
  padding: 12px 24px;
  box-shadow: rgba(0,0,0,0.15) 0 0 0 1px inset;
}

<div
  data-testid="Container"
>
  <button
    class="c0"
  >
    Button
  </button>
  <button
    class="c1"
  >
    Button
  </button>
  <button
    class="c2"
  >
    Button
  </button>
  <button
    class="c1"
  >
    Button
  </button>
  <button
    class="c3"
  >
    Button
  </button>
</div>
`;
